modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/snapshot/CurrencySnapshotCreator.scala

```diff
@@ -50,137 +54,181 @@ trait CurrencySnapshotCreator[F[_]] {
-          blocksForAcceptance,
-          lastContext,
-          lastActiveTips,
-          lastDeprecatedTips,
-          transactions =>
-            rewards
-              .map(_.distribute(lastArtifact, lastContext.snapshotInfo.balances, transactions, trigger, events))
-              .getOrElse(SortedSet.empty[RewardTransaction].pure[F])
-        )
+      def createProposalWithSizeLimit(
+        blocksForAcceptance: List[Signed[Block]],
+        sizeExcessiveBlocks: Set[Signed[Block]] = Set.empty[Signed[Block]]
+      ): F[CurrencySnapshotCreationResult[CurrencySnapshotEvent]] = {
+        for {
+          lastArtifactHash <- lastArtifact.value.hashF
+          currentOrdinal = lastArtifact.ordinal.next
 
-        (awaitingBlocks, rejectedBlocks) = acceptanceResult.notAccepted.partitionMap {
-          case (b, _: BlockAwaitReason)     => b.asRight
-          case (b, _: BlockRejectionReason) => b.asLeft
-        }
+          currentEpochProgress = trigger match {
+            case EventTrigger => lastArtifact.epochProgress
+            case TimeTrigger  => lastArtifact.epochProgress.next
+          }
+          lastActiveTips <- lastArtifact.activeTips
+          lastDeprecatedTips = lastArtifact.tips.deprecated
 
-        (deprecated, remainedActive, accepted) = getUpdatedTips(
-          lastActiveTips,
-          lastDeprecatedTips,
-          acceptanceResult,
-          currentOrdinal
-        )
+          maybeLastDataApplication = lastArtifact.data
 
-        (height, subHeight) <- getHeightAndSubHeight(lastArtifact, deprecated, remainedActive, accepted)
-
-        artifact = CurrencyIncrementalSnapshot(
-          currentOrdinal,
-          height,
-          subHeight,
-          lastArtifactHash,
-          accepted,
-          acceptedRewardTxs,
-          SnapshotTips(
-            deprecated = deprecated,
-            remainedActive = remainedActive
-          ),
-          stateProof,
-          currentEpochProgress,
-          maybeNewDataState
-        )
-      } yield
-        CurrencySnapshotCreationResult(
-          artifact,
-          CurrencySnapshotContext(lastContext.address, snapshotInfo),
-          awaitingBlocks.toSet,
-          rejectedBlocks.toSet
-        )
+          maybeLastDataState <- (maybeLastDataApplication, maybeDataApplicationService).mapN {
+            case (lastDataApplication, (_, service)) =>
+              service
+                .deserializeState(lastDataApplication)
+                .flatTap {
+                  case Left(err) => logger.warn(err)(s"Cannot deserialize custom state")
+                  case Right(a)  => logger.info(s"Deserialized state: $a")
+                }
+                .map(_.toOption)
+                .handleErrorWith(err =>
+                  logger.error(err)(s"Unhandled exception during deserialization data application, fallback to empty state") >>
```

Comment:
Is this something that needs our attention? I.e. if one day we scrape the logs for ERROR messages is this something we need to address, or is it really a WARNing?

Reply:
seems like the comment is pointing the code that has been already merged and the change is only in indentation. can we create a ticket for that and discuss these changes separately @sdavidp ?